Assets/SEE/Controls/Actions/ResizeNodeAction.cs

```diff
@@ -0,0 +1,800 @@
+                    }
+                }
+
+                // Calculate new scale and position
+                Vector3 hitPoint = targetObjectHit.Value.point;
+                Vector3 cursorMovement = Vector3.Scale(currentResizeStep.InitialHitPoint - hitPoint, currentResizeStep.Direction);
+                Vector3 newLocalSize = currentResizeStep.InitialLocalSize - Vector3.Scale(currentResizeStep.LocalScaleFactor, cursorMovement);
+                Vector3 newLocalPosition = currentResizeStep.InitialLocalPosition
+                    - 0.5f * Vector3.Scale(currentResizeStep.LocalScaleFactor, Vector3.Scale(cursorMovement, currentResizeStep.Direction));
+
+                // Contain in parent
+                Bounds2D bounds = new(
+                        newLocalPosition.x - newLocalSize.x / 2,
+                        newLocalPosition.x + newLocalSize.x / 2,
+                        newLocalPosition.z - newLocalSize.z / 2,
+                        newLocalPosition.z + newLocalSize.z / 2
+                );
+                // Parent scale in its own context is always 1
+                Bounds2D otherBounds = new(
+                        -0.5f + currentResizeStep.LocalPadding.x,
+                         0.5f - currentResizeStep.LocalPadding.x,
+                        -0.5f + currentResizeStep.LocalPadding.z,
+                         0.5f - currentResizeStep.LocalPadding.z
+                );
+                if (currentResizeStep.Left && bounds.Left < otherBounds.Left)
+                {
+                    bounds.Left = otherBounds.Left;
+                }
+                if (currentResizeStep.Right && bounds.Right > otherBounds.Right)
+                {
+                    bounds.Right = otherBounds.Right;
+                }
+                if (currentResizeStep.Back && bounds.Back < otherBounds.Back)
+                {
+                    bounds.Back = otherBounds.Back;
+                }
+                if (currentResizeStep.Forward && bounds.Front > otherBounds.Front)
+                {
+                    bounds.Front = otherBounds.Front;
+                }
+
+                // Correct sibling overlap
+                foreach (Transform sibling in siblings)
+                {
+                    Vector3 siblingSize = sibling.gameObject.LocalSize();
+                    Vector3 siblingPos = sibling.localPosition;
+                    // A small offset of 0.0001f is used as a difference between the detection and the set value.
+                    otherBounds.Left  = siblingPos.x - siblingSize.x / 2 - currentResizeStep.LocalPadding.x + 0.0001f;
+                    otherBounds.Right = siblingPos.x + siblingSize.x / 2 + currentResizeStep.LocalPadding.x - 0.0001f;
+                    otherBounds.Back  = siblingPos.z - siblingSize.z / 2 - currentResizeStep.LocalPadding.z + 0.0001f;
+                    otherBounds.Front = siblingPos.z + siblingSize.z / 2 + currentResizeStep.LocalPadding.z - 0.0001f;
```

Comment:
I have added it as:

```cs
            /// <summary>
            /// A small offset of is used as a difference between the detection and the set value
            /// to prevent instant re-detection.
            /// </summary>
            private const float detectionOffset = 0.0001f;
```